### DISCUSSION TITLE
SI Base Units

### DISCUSSION THREAD

**Discussion by thomascamminady** [has2k1/plotnine#832] [2024 Plotnine Contest]
### Authors

Thomas Camminady

### Links

[Github](https://github.com/thomascamminady)

### Full description

# SI Base Units

This graphic attempts to recreate the [NIST SI Base Units poster](https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.1247.pdf) and is simultaneously my first attempt at creating a `plotnine` figure.

I had limited time and limited skills using `plotnine` and doubted whether I should submit my figure in a WIP state but eventually opted to do it.  Basically, I just hope to get a sticker for my laptop ;) 

## Result

![si_units](https://github.com/has2k1/plotnine/assets/24543638/bb4dee0c-66ad-4057-9040-34d1adecdbb4)

## Limitations

Overall, I don't think this is a beautiful visualization, but I liked the work and hope that maybe someone more skilled will continue this. This idea was inspired by the periodic table example in the `plotnine` library.
Comparing this to the original chart, there's a lot of room left for improvements. 

## Data
I turned the original poster into tabular data that can be used as the basis for the chart. See: https://github.com/thomascamminady/si_base_units_relationships_poster_recreate/blob/plotnine-contest-2024/data/units.csv

The data might be helpful if someone else wants to attempt to recreate this plot.

### Code repository

https://github.com/thomascamminady/si_base_units_relationships_poster_recreate/blob/plotnine-contest-2024/notebooks-plotnine-contest-2024/main.ipynb

**Comment by has2k1** [comment]
![si_units](https://github.com/user-attachments/assets/7f1c3992-9aa1-4ad7-9279-bdf8af19569f)

**Comment by EconomiCurtis** [comment]
Ahoy @thomascamminady , 
I am one of the contest organizers. 
If you'd like, we'd be happy to send you a little Plotnine and Posit care package of hex-shaped stickers. Just a small thanks for taking part. https://forms.gle/M8QmM286MSHbkDRd7

**Reply by thomascamminady** [reply]
Thanks so much! Was fun and I learned a lot :)

**Comment by has2k1** [comment]
@thomascamminady, The winning submission has been announced [here](https://posit.co/blog/winner-of-the-2024-plotnine-plotting-contest/). Thank you for taking part in the contest.
